src/Types.hs

```diff
@@ -1188,5 +1194,25 @@ data PurebredEvent
   = NotifyNumThreads Int
                      Generation
   | NotifyNewMailArrived Int
-  | InputValidated (Lens' AppState (Maybe Error))
-                   (Maybe Error)
+  | InputValidated (Lens' AppState (Maybe UserMessage))
+                   (Maybe UserMessage)
```

Comment:
Could you please add comments describing these fields?  At a glance, without reading other code, it seems like this event includes both the message, and a lens to the part of the `AppState` to which the message relates.  And the handler would be responsible for setting it into the `AppState`.  I am not sure if the intuition is correct though.  So we should describe it in more detail.

Reply:
Ok. Done.